### PR TITLE
Check PHP 8.1 compatibility and coding style.

### PR DESCRIPTION
**Pull request by lucaboesch**

### CODE REVIEW

**Comment by lucaboesch** [comment]
Due to changes in strings, I had to cut the CI testing support to Moodle 4.1.
Might be interesting nonetheless though.

File: lib.php [hunk 1]
```diff
@@ -318,7 +316,7 @@ function mod_subcourse_cm_info_view(cm_info $cm) {
         $grades = grade_get_grades($cm->course, 'mod', 'subcourse', $cm->instance, $USER->id);
         $currentgrade = (empty($grades->items[0]->grades)) ? null : reset($grades->items[0]->grades);
 
-        if (($currentgrade !== null) and isset($currentgrade->grade) and !($currentgrade->hidden)) {
+        if (($currentgrade !== null) && isset($currentgrade->grade) && !($currentgrade->hidden)) {
```
Comment (joaop221) [inline lib.php:319]:
```suggestion
        if ($currentgrade !== null && isset($currentgrade->grade) && !$currentgrade->hidden) {
```

**Review by joaop221 — APPROVED** [review]

**Comment by joaop221** [comment]
Is there any plan to complete this feature?

**Comment by danmarsden** [comment]
@joaop221 we've just taken over maintenance of this plugin this week from David Mudrak - I haven't had a chance to get into all the issues in the backlog but getting support for PHP 8 and latest versions of Moodle is near the top of my list - if you have internal development capabilities, pull requests are always welcome too!

**Pull request merged**